packages/storage/src/service.ts

```diff
@@ -17,55 +17,138 @@
 
 import { FirebaseApp } from '@firebase/app-types';
 import * as args from './implementation/args';
-import { AuthWrapper } from './implementation/authwrapper';
 import { Location } from './implementation/location';
-import * as RequestExports from './implementation/request';
+import { FailRequest } from './implementation/failrequest';
+import { Request, makeRequest } from './implementation/request';
+import { RequestInfo } from './implementation/requestinfo';
 import { XhrIoPool } from './implementation/xhriopool';
 import { Reference } from './reference';
 import { Provider } from '@firebase/component';
-import { FirebaseAuthInternalName } from '@firebase/auth-interop-types';
+import {
+  FirebaseAuthInternalName,
+  FirebaseAuthTokenData
+} from '@firebase/auth-interop-types';
+import { FirebaseOptions } from '@firebase/app-types-exp';
+import * as constants from '../src/implementation/constants';
+import { RequestMap } from './implementation/requestmap';
+import { requestMaker } from './implementation/requestmaker';
+import * as errorsExports from './implementation/error';
 
 /**
  * A service that provides firebaseStorage.Reference instances.
  * @param opt_url gs:// url to a custom Storage Bucket
  *
  * @struct
  */
-export class Service {
-  authWrapper_: AuthWrapper;
-  private app_: FirebaseApp;
+export class StorageService {
+  private app_: FirebaseApp | null;
   private bucket_: Location | null = null;
   private internals_: ServiceInternals;
+  private authProvider_: Provider<FirebaseAuthInternalName>;
+  private appId_: string | null = null;
+
+  private storageRefMaker_: (loc: Location) => Reference;
+  private requestMaker_: requestMaker;
+  private pool_: XhrIoPool;
+  private requestMap_: RequestMap;
+  private deleted_: boolean = false;
+  private maxOperationRetryTime_: number;
+  private maxUploadRetryTime_: number;
```

Comment:
Should we mark these as readonly (where possible)?

Reply:
Sure, done.